src/components/registration/Eligibility.tsx

```diff
@@ -1,97 +1,184 @@
+          formId={'androidVerify'}
+          onSubmit={(event: any) => {
+            const selectedOption = event.formData.android_verify
+            if (selectedOption && Object.keys(selectedOption).length === 0)
+              setErrorMessage('You must choose an option')
+            else {
+              setChecks(prev => {
+                return { ...prev, hasAndroid: selectedOption.has_android }
+              })
+              setStep((current: number) =>
+                current < MAX_STEPS ? current + 1 : current,
+              )
             }
-            label={t('eligibility.text4')}
-          />
-        </div>
+          }}
+        />
       </div>
+    )
+  }
 
-      <div className="text-center">
-        <Button
-          color="primary"
-          variant="contained"
-          size="medium"
-          type="submit"
-          disabled={Object.values(checks).some(value => value !== true)}
-          className="wideButton"
-          onClick={() => setEligibilityFn()}
-        >
-          {t('common.continue')}
-        </Button>
+  if (step === 4) {
+    return (
+      <div className="quizWrapper">
+        <ProgressBar step={step} maxSteps={MAX_STEPS} />
+        <SageForm
+          title={
+            'Have you ever felt you could have benefited / did benefit from access to support for anxiety or depression?'
+          }
+          errorMessage={errorMessage}
+          formId={'supportVerify'}
+          onSubmit={(event: any) => {
+            const selectedOption = event.formData.support_verify
+            if (selectedOption && Object.keys(selectedOption).length === 0)
+              setErrorMessage('You must choose an option')
+            else {
+              setChecks(prev => {
+                return { ...prev, supportSurvey: selectedOption.accept }
+              })
+              setStep((current: number) =>
+                current < MAX_STEPS ? current + 1 : current,
+              )
+            }
+          }}
+        />
       </div>
-    </div>
-  )
+    )
+  }
+
+  if (step === 5) {
+    if (
+      checks.hasAndroid &&
+      checks.inValidLocation &&
+      checks.supportSurvey &&
+      checks.validAgeRange
+    )
+      setEligibilityFn()
+
+    return (
+      <div className="quizWrapper">
+        <div className="headerWrapper">
+          <h1>Thank you for your time!</h1>
+        </div>
+        <Separator />
+        <div className="rejectionText">
+          It looks like this is not the right study for you, but we really
+          appreciate your interest in helping us out!
+        </div>
+        <NavLink to="/home">
+          <Button
+            color="primary"
+            variant="contained"
+            size="large"
+            className="wideButton"
+            onClick={() => {
+              return <Redirect to="/home" />
+            }}
+          >
+            Back to home
+          </Button>
+        </NavLink>
+      </div>
+    )
+  }
+
+  return <></>
```

Comment:
I'm curious, is there a difference in returning empty fragment vs null ?

Reply:
I think TS could make you do things like returning a fragment instead of null depending on how it was written, so far it's done that way in most of the covid project. Nevertheless I think it could be a tiny bit more performant to just return null, I'll change that null and consider it for the rest of the cases.